src/python/ensembl/io/genomio/genome_stats/compare.py

```diff
@@ -44,53 +52,47 @@ def _diff_dicts(ncbi: Dict[str, int], core: Dict[str, int]) -> Dict[str, Any]:
         if ncbi_count == core_count:
             if ncbi_count != 0:
                 same[key] = ncbi_count
-            continue
-        diff[key] = {"ncbi": ncbi_count, "core": core_count, "diff": core_count - ncbi_count}
-
-    comp: Dict[str, Any] = {}
-    if same:
-        comp["same"] = same
-    if diff:
-        comp["different"] = diff
-    return comp
+        else:
+            diff[key] = {"ncbi": ncbi_count, "core": core_count, "diff": core_count - ncbi_count}
+    return {"same": same, "different": diff}
```

Comment:
I was not 100% about this change I made to be honest, felt that the structure consistency would be good, but Python has different ways to handle missing keys in a readable way. Furthermore, with unit tests it means it is highly unlikely that a missing key may mean something was missed in the processing, so this can be dropped.